bears/python/YapfBear.py

```diff
@@ -0,0 +1,137 @@
+import sys
+
+from coalib.bears.LocalBear import LocalBear
+from yapf.yapflib.yapf_api import FormatFile
+from coalib.results.Result import Result
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from coalib.results.Diff import Diff
+from coalib.misc.ContextManagers import prepare_file
+from coalib.bears.requirements.PipRequirement import PipRequirement
+
+
+class YapfBear(LocalBear):
+    """
+    Check and correct formatting of Python Code using ``yapf`` utility.
+
+    See <https://github.com/google/yapf> for more information.
+    """
+    LANGUAGES = {"Python"}
+    AUTHORS = {'The coala developers'}
+    REQUIREMENTS = {PipRequirement('yapf', '0.10')}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+
+    # TODO Add coalesce_brackets on supported by yapf
+    def run(self, filename, file,
+            max_line_length: int=79,
+            tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
+            allow_multiline_lambdas: bool=False,
+            blank_line_before_nested_class_or_def: bool=False,
+            continuation_tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
+            dedent_closing_brackets: bool=False,
+            indent_dictionary_value: bool=False,
+            join_multiple_lines: bool=True,
+            spaces_around_power_operator: bool=False,
+            spaces_before_comment: int=2,
+            space_between_ending_comma_and_closing_bracket: bool=True,
```

Comment:
where? because if its for tuples, we want it False, as ``((tuple_member_1, tuple_member_2),)`` is a 1 element tuple